.github/workflows/Build_Wheels_Upload_PyPI.yml

```diff
@@ -0,0 +1,168 @@
+# and uploading them to the PyPI package index.
+#
+# v0.1, Michael Wuertenberger
+#
+
+name: Build_Wheels_Upload_PyPI
+
+on:
+  push:
+    # Run this action on the trigger event when a tag is pushed
+    tags:
+       - '*'
+    # Other setups if pushes in master only if setup changed or the workflow
+    # itself. As the version number is generated automatically, this might
+    # not make sense.
+
+    # branches:
+    #   - master
+    # paths:
+    #   - 'setup.py'
+    #   - '.github/workflows/Build_Wheels_Upload_PyPI.yml'
+
+jobs:
+
+  # This action is split into three jobs:
+  # - Building the distribution
+  # - Building the wheels for the distribution
+  # - Uploading the artifacts to PyPI package
+  # The first and second job run in parallel.
+  # The uploading jos needs to have the other two finished without error.
+
+  build_sdist:
+
+    # First the distribution is done on ubuntu. This is not related to any
+    # operating system, so we could do it on the default os og GitHub.
+    # Python 3.9 is chosen as it is the actual version your could work with.
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: install_python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
+
+      # For the build, sep needs numpy and cython and I add twine and wheel
+      # for better testing and checking.
+
+      - name: install_dependencies
+        run: python -m pip install twine numpy wheel cython
+
+      - name: build_sdist
+        run: python setup.py sdist
+
+      - name: show_files
+        run: ls -lh dist
+        shell: bash
+
+      - name: check_metadata
+        run: twine check dist/*
+
+      - name: upload_sdist
+        uses: actions/upload-artifact@v2
+        with:
+          name: dist
+          path: dist/*.tar.gz
+
+  build_wheels:
+
+    # Second the wheels are build for different OS and python versions. This is
+    # done with the help of <cibuildwheel> package. Again, beside this package
+    # numpy, cython is needed, adding certify and wheel for build.
+    #
+    # The wheels are build for Windows, Linux and MacOS and the python versions
+    # 3.5 - 3.9
+    # Please check if there are constraints of the manylinux configuration I
+    # have chosen so far! This is related to the binary package you are
+    # developing.
+    # The three operating system could be done in parallel.
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 3
+      matrix:
+        python-version: [3.9]
+        os: [windows-latest, macos-latest, ubuntu-latest]
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+
+      - name: Setup_Python_${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: install_deps
+        run: python -m pip install cibuildwheel wheel cython numpy certifi
```

Comment:
Do we actually need to install `wheel cython numpy`? These are declared as build-time dependencies in `pyproject.toml`.

Reply:
Kyle,
I tried it without NumPy and Cython, but it did not work. I put in all packages, the build process demanded.
My guess is the some of the interface headers are used.

Some observations: There is a warning about the interface version to NumPy, which might cause trouble in future. This might lead to set a defined NumPy version for building the wheels or even better to work on the warning.
Second I see a lot of sprint warnings and the hint to replace them with sprintf_s an so on.

Michel

Reply:
It is indeed necessary to have numpy and cython installed for "building" the source distribution with `python setup.py sdist` (it runs setup.py directly). For building wheels via a pip command, I believe pip first reads `pyproject.toml` and installs the build time dependencies. I removed numpy and cython from the wheel build step and it seems to work.

The `pyproject.toml` declares `oldest-supported-numpy` as the build-time dependency, so we do use a fixed numpy version when building, depending on the Python version. Not using the oldest supported numpy when building may have been the cause of the interface version warnings you were seeing, but I'm not sure.

I'm not sure what the cause of the sprint warnings are. Maybe open an issue for that if you think there's something actionable we can fix.

Thanks again for this PR... it's super super cool to be able to just push a tag to do a full release!